examples/cuda/test_profiler.cpp

```diff
@@ -0,0 +1,343 @@
+// Copyright (c) 2020, University of Washington All rights reserved.
```

Comment:
I was looking for an example to initialize the profiler. I'd imagine this file might've to do that, but couldn't a call that does that... how does this test initialize profiler?

Reply:
It happens internally: https://github.com/bespoke-silicon-group/bsg_replicant/blob/18d4ccdcc88dbc6e482bef2bf318bcf87f799dbc/libraries/platforms/aws-fpga/bsg_manycore_platform.cpp#L458

Reply:
Okay. So, turned on by default?

Reply:
Yeah. It is already. This hooks into the `vanilla_core_profiler` module, not `vanilla_core_trace`

Reply:
I makes sense. That, host code wouldn't have to worry about managing profiler but can just get info by calling get_icount etc. at will.

Reply:
Right. And if it isn't implemented it'll return NO_IMPL. 

Though my motivation was somewhat more mundane: It avoids `ifdef COSIM`

Reply:
Since this profiler actual hardware, there's no benefit to stopping it also ;)